examples/transmission_main_example.py

```diff
@@ -77,10 +61,63 @@ def main(args):
     axis('off')
     show()
 
+
+def main(args):
+    input_filename = str(args.filename)
+    split_filename = input_filename.split(".")
+    json_filename = split_filename[0]+'.json'
+    try:
+        assert split_filename[1] in ('json','xls','csv','xlsm')
+    except AssertionError as e:
+        print(f'invalid file extension .{split_filename[1]}')
+        raise e
+    if split_filename[1] != 'json':
+        print(f'parse excel input to {json_filename}')
+        convert_file(input_filename)
+
+    json_data = load_json(json_filename)
+    read_eqpt_library(EQPT_LIBRARY_FILENAME)
+
+    network = network_from_json(json_data)
+    build_network(network)
+
+    spacing = 0.05 #THz
+    si = SpectralInformation() # !! SI units W, Hz
+    si = si.update(carriers=tuple(Channel(f, (191.3+spacing*f)*1e12, 
+            32e9, 0.15, Power(1e-3, 0, 0)) for f in range(1,97)))
+
+    trx = [n for n in network.nodes() if isinstance(n, Transceiver)]
+    if args.list>=1:
+        print(*[el.uid for el in trx], sep='\n')
+    else:
+        try:
+            source = next(el for el in trx if el.uid == args.source)
+        except StopIteration as e:
+            source = trx[0]
+            print(f'invalid souce node specified: {args.source!r}, replaced with {source}')
```

Comment:
Replace `print` to `stdout` with either `logging` or `warning`
https://docs.python.org/3/library/logging.html
https://docs.python.org/3/library/warnings.html

Reply:
yes I know: hence the #TODO logging in several parts of my code, but one thing at a time...
can you do it?